### PR TITLE
Allow nested Sidekiq::Testing

### PR DESCRIPTION
**Pull request by cjubb39**
Hello!

With the upgrade from version `7.1.1` -> `7.1.6`, I started seeing failures on having nested `Sidekiq::Testing.inline!` blocks.  The `inline` status seemed to be lost after the inner block exited.  This behavior change seemed to be introduced in e9f212902185591831af514df6313f8ae2cce31a.  I've included a test that fails on `main` but passes with my change.

I'm a bit of a Ruby noob so happy for whatever stylistic changes :)

### CODE REVIEW

**Comment by mperham** [comment]
I don’t want to support reentrancy. Nesting fake and inline modes makes it hard to follow what the code will actually do.

**Comment by cjubb39** [comment]
Fair enough.  In that case, are you open to a change that raises some sort of error on nesting?

**Comment by mperham** [comment]
I think that’s reasonable.

**Comment by cjubb39** [comment]
Closing in favor of #6078
